### PR TITLE
Tools: Add Feature - Add script to generate system config info.

### PR DESCRIPTION
**Pull request by yukirora**
**Description**
Add script to generate system config info.

**Major Revision**
- Add script to generate system config info into the dict in superbench/tools.

### CODE REVIEW

File: superbench/tools/system_info.py [hunk 1]
```diff
@@ -0,0 +1,442 @@
...
+            is_selected = True
+            if selected is not None:
+                is_selected = False
+                for key in selected:
+                    if key in line:
+                        is_selected = True
+            if not is_selected:
+                i += 1
+                continue
+            # process with indent recursively
```
Comment (cp5555) [inline superbench/tools/system_info.py:85]: Cannot understand the logic clearly. Would you please give an example for this recursive parsing?

Comment (yukirora) [reply superbench/tools/system_info.py:85]: I will do the recursive parsing to deal with the different indent in content into different level dict, for example, the content is as the following, if next line has more indent compared to current line, I will keep this line as a key, and then recursively parse the lines which have the same indent with next line as the value of the key.
```
CA 'mlx5_0'
        CA type: MT4123
        Number of ports: 1
        Firmware version: 20.30.1004
        Hardware version: 0
        Node GUID: 0x88e9a4ffff0a28f4
        System image GUID: 0x88e9a4ffff0a28f4
        Port 1:
                State: Down
                Physical state: Disabled
                Rate: 40
                Base lid: 0
                LMC: 0
                SM lid: 0
                Capability mask: 0x00010000
                Port GUID: 0x8ae9a4fffe0a28f4
                Link layer: Ethernet
CA 'mlx5_1'
        CA type: MT4123
        Number of ports: 1
        Firmware version: 20.30.1004
        Hardware version: 0
        Node GUID: 0x88e9a4ffff0a28f5
        System image GUID: 0x88e9a4ffff0a28f4
        Port 1:
                State: Active
                Physical state: LinkUp
                Rate: 200
                Base lid: 38
                LMC: 0
                SM lid: 1
                Capability mask: 0x2651e848
                Port GUID: 0x88e9a4ffff0a28f5
                Link layer: InfiniBand
```
so the result of the parsing will be like as the following:
```
                "CA 'mlx5_0'": {
                    "CA type": "MT4123",
                    "Number of ports": "1",
                    "Firmware version": "20.30.1004",
                    "Hardware version": "0",
                    "Node GUID": "0x88e9a4ffff0a28f4",
                    "System image GUID": "0x88e9a4ffff0a28f4",
                    "Port 1:": {
                        "State": "Down",
                        "Physical state": "Disabled",
                        "Rate": "40",
                        "Base lid": "0",
                        "LMC": "0",
                        "SM lid": "0",
                        "Capability mask": "0x00010000",
                        "Port GUID": "0x8ae9a4fffe0a28f4",
                        "Link layer": "Ethernet"
                    }
                },
                "CA 'mlx5_1'": {
                    "CA type": "MT4123",
                    "Number of ports": "1",
                    "Firmware version": "20.30.1004",
                    "Hardware version": "0",
                    "Node GUID": "0x88e9a4ffff0a28f5",
                    "System image GUID": "0x88e9a4ffff0a28f4",
                    "Port 1:": {
                        "State": "Active",
                        "Physical state": "LinkUp",
                        "Rate": "200",
                        "Base lid": "38",
                        "LMC": "0",
                        "SM lid": "1",
                        "Capability mask": "0x2651e848",
                        "Port GUID": "0x88e9a4ffff0a28f5",
                        "Link layer": "InfiniBand"
                    }
                },
```

File: superbench/tools/system_info.py [hunk 2]
```diff
@@ -0,0 +1,442 @@
...
+            prefix + command,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT,
+            shell=True,
+            check=False,
+            universal_newlines=True
+        )
+        return output.stdout
+
+    def __cal_indent(self, line, symbol='\t'):
```
[the remark this answers is not in the document]
Comment (yukirora) [reply superbench/tools/system_info.py:37]: I only want to count the symbol at the beginning of the content, for example, '/t/txxxx/txxxx', the result should be 2 , [string].count('charater') not work for me.

**Review by cp5555 — APPROVED** [review]

File: setup.py [hunk 3]
```diff
@@ -142,6 +142,7 @@ def run(self):
         'natsort>=7.1.1',
         'omegaconf==2.0.6',
         'pyyaml>=5.3',
+        'xmltodict',
```
Comment (abuccts) [inline setup.py:145]: add a minimum version?

File: superbench/tools/system_info.py [hunk 4]
```diff
@@ -0,0 +1,447 @@
...
+
+import json
+import subprocess
+import xmltodict
+from pathlib import Path
+
+
+class SystemInfo():
+    """Systsem info class."""
+    def __run_cmd(self, command, non_root=False):
```
Comment (abuccts) [inline superbench/tools/system_info.py:14]: non_root=False -> root=True?

File: superbench/tools/system_info.py [hunk 5]
```diff
@@ -0,0 +1,447 @@
...
+        Args:
+            command (string): the command to run in terminal.
+            non_root (bool): the mode to run the command, root or non-root.
+
+        Returns:
+            string: the stdout string of the command.
+        """
+        prefix = ''
+        if not non_root:
+            prefix = 'sudo '
```
Comment (abuccts) [inline superbench/tools/system_info.py:26]: what if `sudo` is not installed? and usually there's password for `sudo`

File: superbench/tools/system_info.py [hunk 6]
```diff
@@ -0,0 +1,447 @@
...
+
+    def get_cpu(self):
+        """Get CPU info.
+
+        Returns:
+            dict: cpu info dict.
+        """
+        lscpu_dict = {}
+        try:
+            # get general cpu infomation from lscpu
```
Comment (abuccts) [inline superbench/tools/system_info.py:163]: infomation -> information

File: superbench/tools/system_info.py (hunk 5 above)
Comment (abuccts) [reply superbench/tools/system_info.py:26]: seems non_root is not used, can you remove `sudo` here and let others run `sudo python system_info.py` directly?

File: superbench/tools/system_info.py (hunk 6 above)
Comment (yukirora) [reply superbench/tools/system_info.py]: fixed

File: superbench/tools/system_info.py (hunk 5 above)
Comment (yukirora) [reply superbench/tools/system_info.py]: ok

File: setup.py (hunk 3 above)
Comment (yukirora) [reply setup.py]: added

**Review by abuccts — APPROVED** [review]

**Pull request merged**